modules/tinymce/src/core/test/ts/browser/UndoManagerTest.ts

```diff
@@ -682,4 +683,25 @@ describe('browser.tinymce.core.UndoManagerTest', () => {
     });
   });
 
+  context('TINY-9222: Scroll to the cursor after undo or redo actions', () => {
+    ([ 'undo', 'redo' ] as const).forEach((action) => {
+      it('TINY-9222: Scroll to the cursor after ' + action, () => {
+        const editor = hook.editor();
+        editor.setContent(`<p class="first">top paragraph</p>${'<br>'.repeat(50)}<p class="last">last paragraph</p>`);
+        TinySelections.select(editor, 'p.last', [ 0 ]);
+        TinyContentActions.type(editor, 'updated ');
+
+        const doc = TinyDom.document(editor);
+        Scroll.to(0, 0, doc);
+        editor.undoManager[action]();
```

Comment:
on `redo` action phase: how do you test `redo` without the previous `undo`? It seems strange here that `redo` after the update scrolls content down since nothing to redo.

Reply:
Yeah, maybe do both in one test. Make a change, undo, check scroll pos, redo check scroll pos or something.

Reply:
Yeah, that makes sense. However, I believe that the `undoManager` state was persistent and `redo` actually worked! Weird. Updated to a single test [here](https://github.com/tinymce/tinymce/pull/8205/commits/d03dcd64f8958af59ee6b7984f043033afb135d8)